domain/users/user_dto.go

```diff
@@ -0,0 +1,67 @@
+package users
+
+import (
+	"moku-moku/utils/errors"
+	"regexp"
+	"strings"
+)
+
+type User struct {
+	Email      string `json:"email"`
+	Username   string `json:"username"`
+	Name       string `json:"name"`
+	Surname    string `json:"surname"`
+	Biography  string `json:"biography"`
+	Birthday   string `json:"birthday"`
+	Password   string `json:"-"`
+	PasswordR  string `json:"-"`
+	ProfilePic string `json:"profile_picture"`
+	Points     int32  `json:"points"`
+}
+
+func (user *User) EmailValidation() *errors.RestErr {
+	// Email sanitation
+	user.Email = strings.TrimSpace(strings.ToLower(user.Email))
+
+	if user.Email == "" {
+		return errors.BadRequest("invalid email address")
+	}
+
+	// Email validation
+	emailRegex := regexp.MustCompile(`^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$`)
```

Comment:
May be currently there are domains like .online and .website that have more than 4 chars so shall we update the regex like `...[\w-]{2,7}$`?

Reply:
Yes! We can implement that, the commit message could be somethinf like -> feat: Support for 2-7 char domain email